rust/agama-lib/src/utils/transfer/file_systems.rs

```diff
@@ -0,0 +1,284 @@
+// Copyright (c) [2025] SUSE LLC
+//
+// All Rights Reserved.
+//
+// This program is free software; you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 2 of the License, or (at your option)
+// any later version.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, contact SUSE LLC.
+//
+// To contact SUSE LLC about this file by physical or electronic mail, you may
+// find current contact information at www.suse.com.
+
+/// Module to search for file systems.
+use std::{path::PathBuf, process::Command};
+
+use regex::Regex;
+
+/// Represents a file system from the underlying system.
+///
+/// It only includes the elements that are relevant for the transfer API.
+#[derive(Clone, Debug, Default)]
+pub struct FileSystem {
+    pub block_device: String,
+    pub fstype: Option<String>,
+    pub mount_point: Option<PathBuf>,
+    pub transport: Option<String>,
+    pub label: Option<String>,
+}
+
+impl FileSystem {
+    /// Whether the file system was mounted.
+    pub fn is_mounted(&self) -> bool {
+        self.mount_point.is_some()
+    }
+
+    /// Kernel name of the block device containing the file system.
+    pub fn device(&self) -> String {
+        format!("/dev/{}", &self.block_device)
+    }
+
+    /// Whether the file system can be mounted.
+    ///
+    /// File systems that cannot be mounted are ignored.
+    fn can_be_mounted(&self) -> bool {
+        let Some(fstype) = &self.fstype else {
+            return false;
+        };
+
+        match fstype.as_str() {
+            "" | "crypto_LUKS" | "swap" => false,
+            _ => true,
+        }
+    }
+}
+
+/// Holds a list of file systems.
+///
+/// It offers a set of convenience method to search within the list.
+#[derive(Debug, Default)]
+pub struct FileSystemsList {
+    file_systems: Vec<FileSystem>,
+}
+
+impl FileSystemsList {
+    /// Creates a list of file systems.
+    pub fn new(file_systems: Vec<FileSystem>) -> Self {
+        Self { file_systems }
+    }
+
+    /// Creates a list for the file systems in the underlying system.
+    pub fn from_system() -> Self {
+        let file_systems = FileSystemsReader::read_from_system();
+        Self::new(file_systems)
+    }
+
+    pub fn to_vec(&self) -> Vec<FileSystem> {
+        self.file_systems.clone()
+    }
+
+    /// Returns the file system with the given block device name.
+    ///
+    /// * `name`: block device name.
+    pub fn by_name(&self, name: &str) -> Option<&FileSystem> {
```

Comment:
regarding naming I would probably use `with_` prefix instead of `_by` as by invokes to me sorted "by"...and with invokes for me objects "with".
Also what I do not like is inconsistent return type. So I would probably use
```
find_name
with_label
with_transport
```